txp/Pos/Txp/Toil/Logic.hs

```diff
@@ -180,14 +182,19 @@ verifyGState curEpoch txAux txFee = do
 -- stakeholders. Notice: it doesn't use actual txdistr type because
 -- it's defined in txp module above.
 bootRelatedDistr :: GenesisWStakeholders -> [(StakeholderId, Coin)] -> Bool
-bootRelatedDistr (GenesisWStakeholders bootHolders) txOutDistr =
-    -- All addresses in txDistr are from bootHolders and every boot
-    -- stakeholder is mentioned in txOutDistr
-    getKeys bootHolders == HS.fromList stakeholders &&
-    -- Every stakeholder gets his divisor. It's safe to use ! here
-    -- because we're already sure that bootHolders ~ addrs is
-    -- bijective.
-    all (\(a,c) -> c >= (minimumPerStakeholder HM.! a)) txOutDistr
+bootRelatedDistr g@(GenesisWStakeholders bootHolders) txOutDistr
+    -- We allow small outputs to attribute stake to whoever because
+    -- it's not likely they will shift overall stake distribution
+    -- much.
+    | coinSum < unsafeGetCoin (bootDustThreshold g) = True
```

Comment:
For me it makes more sense to only allow genesis stakeholders in distribution (but not necessary all genesis stakeholders).